core/storage/base_model/gae_models.py

```diff
@@ -380,11 +379,7 @@ class BaseCommitLogEntryModel(BaseModel):
     """Base Model for the models that store the log of commits to a
     construct.
     """
-    # Update superclass model to make these properties indexed.
-    created_on = ndb.DateTimeProperty(auto_now_add=True, indexed=True)
```

Comment:
As an FYI the documentation for the corresponding properties in the superclass seems wrong (re what is allowable with regards to setting it automatically/directly). Could you fix them please? Try to be prescriptive, i.e. specify what callers are expected to do with it. Maybe also do an audit to ensure that these are all being handled correctly so we don't have more cases like this...

Reply:
Please respond to this comment (especially the last part).

Reply:
Updated the comment. I did the audit and that led to the refactor of `put` methods, nothing else seemed broken.